prov/util/src/util_mem_monitor.c

```diff
@@ -194,6 +194,21 @@ int ofi_monitors_add_cache(struct ofi_mem_monitor **monitors,
 			ret = monitor->start(monitor);
 			if (ret) {
 				cache->monitors[iface] = NULL;
+
+				/* Requested memory monitor is not supported.
+				 * The cache cannot be used for this HMEM type.
+				 */
+				if (ret == -FI_ENOSYS) {
+					FI_DBG(&core_prov, FI_LOG_MR,
+					       "MR cache not supported for %s memory\n",
+					       fi_tostr(&iface, FI_TYPE_HMEM_IFACE));
+					ret = FI_SUCCESS;
```

Comment:
No need for this assignment.  (Static code analysis might complain anyway that the value can be overridden before being used.)  Simply continue here.

Reply:
I'm not sure I can do this. ret is returned to the user. It needs to be updated for this cause (or a different variable used).

Reply:
I would restructure to:
```
ret = monitor->start(monitor);
if (ret && ret != -FI_ENOSYS)
    goto err;
```
then add an error case below:
```
err:
    unlock();
    FI_WARN(...)
    ofi_monitors_del_cache(cache);
    return ret;
```
This avoids a duplicate check for an error once one has already been found.  The common exit path then just becomes:
```
return success_count ? FI_SUCCESS : -FI_ENOSYS
```
unless you want to keep the FI_DBG message.

I don't think setting cache->monitors[iface] = NULL is needed, since that's done at the top of the function anyway.